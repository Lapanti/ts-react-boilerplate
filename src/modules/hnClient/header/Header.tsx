import * as React from 'react';
import { connect } from 'react-redux';
import { NavLink } from 'react-router-dom';

import styled from '../../../theme/styled';

const StyledHeader = styled.header`
  align-items: center;
  background-color: ${props => props.theme.primaryColor};
  display: flex;
  justify-content: flex-start;
`;

const Title = styled.h1`
  font-weight: bold;
  font-size: 2rem;
  margin: 0 2rem;
`;

const Category = styled(NavLink).attrs({
  activeStyle: {
    fontWeight: 'bold',
  },
})`
  color: black;
  cursor: pointer;
  font-size: 1.6rem;
  margin: 0 1rem;
  text-decoration: none;
`;

export const Header: React.StatelessComponent<{}> = () => (
  <StyledHeader>
    <Title>HN PWA</Title>
    <Category to="/new">New</Category>
    <Category to="/top">Top</Category>
    <Category to="/best">Best</Category>
  </StyledHeader>
);

export default connect()(Header);
