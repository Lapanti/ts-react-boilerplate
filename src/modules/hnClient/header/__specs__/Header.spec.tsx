import * as React from 'react';
import { shallow } from 'enzyme';
import createHistory from 'history/createBrowserHistory';

import { Header } from '../Header';

describe('<Header />', () =>
  it('should render with correct props', () =>
    expect(shallow(<Header />)).toMatchSnapshot()));
