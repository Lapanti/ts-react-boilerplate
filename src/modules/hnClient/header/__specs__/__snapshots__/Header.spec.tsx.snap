// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render with correct props 1`] = `
<styled.header>
  <styled.h1>
    HN PWA
  </styled.h1>
  <Styled(NavLink)
    to="/new"
  >
    New
  </Styled(NavLink)>
  <Styled(NavLink)
    to="/top"
  >
    Top
  </Styled(NavLink)>
  <Styled(NavLink)
    to="/best"
  >
    Best
  </Styled(NavLink)>
</styled.header>
`;
